ste/xfer-anyToRemote-folder.go

```diff
@@ -0,0 +1,133 @@
+// Copyright © 2017 Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ste
+
+import (
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-azcopy/common"
+)
+
+// anyToRemote_folder handles all kinds of sender operations for FOLDERs - both uploads from local files, and S2S copies
+func anyToRemote_folder(jptm IJobPartTransferMgr, info TransferInfo, p pipeline.Pipeline, pacer pacer, senderFactory senderFactory, sipf sourceInfoProviderFactory) {
+
+	// step 1. perform initial checks
+	if jptm.WasCanceled() {
+		jptm.ReportTransferDone()
+		return
+	}
+
+	// step 2a. Create sender
+	srcInfoProvider, err := sipf(jptm)
+	if err != nil {
+		jptm.LogSendError(info.Source, info.Destination, err.Error(), 0)
+		jptm.SetStatus(common.ETransferStatus.Failed())
+		jptm.ReportTransferDone()
+		return
+	}
+	if srcInfoProvider.EntityType() != common.EEntityType.Folder() {
+		panic("configuration error. Source Info Provider does not have Folder entity type")
+	}
+
+	sBase, err := senderFactory(jptm, info.Destination, p, pacer, srcInfoProvider)
+	if err != nil {
+		jptm.LogSendError(info.Source, info.Destination, err.Error(), 0)
+		jptm.SetStatus(common.ETransferStatus.Failed())
+		jptm.ReportTransferDone()
+		return
+	}
+	s := sBase.(IFolderSender)
+
+	/*TODO
+	// step 3: check overwrite option
```

Comment:
I imagine this is to be done soon :tm: as well? (as in not this PR but another)

Reply:
Oh, it was supposed to be in here!  Let me check. Thanks for spotting it.

Reply:
Correction. It's done. I just forgot to remove the old TODO.  

It's done in simpleFolderTracker.ShouldSetProperties